src/sentry/notifications/notifications/organization_request/integration_request.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Iterable, Mapping, MutableMapping, Sequence
+
+from sentry.notifications.notifications.organization_request import OrganizationRequestNotification
+from sentry.notifications.utils.actions import MessageAction
+from sentry.utils.http import absolute_uri
+
+if TYPE_CHECKING:
+    from sentry.models import Organization, Team, User
+
+
+def get_url(organization: Organization, provider_type: str, provider_slug: str) -> str:
+    # Explicitly typing to satisfy mypy.
+    url: str = absolute_uri(
+        "/".join(
+            [
+                "/settings",
+                organization.slug,
+                {
+                    "first_party": "integrations",
+                    "plugin": "plugins",
+                    "sentry_app": "sentry-apps",
+                }.get(provider_type, ""),
+                provider_slug,
+                "?referrer=request_email",
```

Comment:
@mgaeta we have the `get_referrer` function we can use here instead

Reply:
It looks like `get_url` is used for slack which makes the `request_email` referrer incorrect. The integration provider should be part of the referrer so we know which notification folks came from when landing on the page.